arrow-ord/src/sort.rs

```diff
@@ -724,15 +724,46 @@ pub fn lexsort_to_indices(
         len = limit.min(len);
     }
 
-    let lexicographical_comparator = LexicographicalComparator::try_new(columns)?;
-    // uint32 can be sorted unstably
-    sort_unstable_by(&mut value_indices, len, |a, b| {
+    match columns.len() {
```

Comment:
I think it would be worth a comment here explaining the repetition is purposeful to generate specialized code for small numbers of columns

```suggestion
    // Instantiate specialized versions of comparisons for small numbers
    // of columns as it helps the compiler generate better code.
    match columns.len() {
```